### PR TITLE
GraphQl-622: No possibility to update customizable_options in updateCartItems mutation

### PR DESCRIPTION
**Pull request by rogyar**
### Description (*)
This PR introduces a possibility to update customisable options for cart items using `updateCartItems` mutation. 

### Fixed Issues (if relevant)

1. #622: Add possibility to update customizable_options in updateCartItems mutation

### Manual testing scenarios (*)
1. Create empty cart
2. Add a product to the shopping cart with customizable options
3. Use `updateCartItems` mutation to update the customizable options in the following way

```graphql
mutation {
  updateCartItems(input: {
    cart_id:"$maskedCartId"
    cart_items: [
      {
        cart_item_id:23
        quantity:-1
        customizable_options: [
          {
            id:1
            value:"test"
          }
        ]
      }
    ]
  }) {
    cart {
      items {
        qty
        product {
          name
        }
        ... on SimpleCartItem {
          customizable_options {
            label
            values {
              label
              value
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

**Review by naydav — CHANGES_REQUESTED** [review]
@rogyar, please, align code to last changes in mainline
thanks
Also, need to cover functionality with API-functional tests

If you would like you can create a separate ticket for this

**Comment by rogyar** [comment]
I'm currently working on the tests. Thank you

File: app/code/Magento/QuoteGraphQl/Model/Resolver/UpdateCartItems.php [hunk 1]
```diff
@@ -94,53 +111,47 @@ private function processCartItems(Quote $cart, array $items): void
...
-     * @return void
-     * @throws GraphQlInputException
+     * @param int $itemId
+     * @param Quote $cart
+     * @param float $qty
+     * @throws GraphQlNoSuchEntityException
+     * @throws NoSuchEntityException
+     * @throws GraphQlNoSuchEntityException
      */
-    private function validateCartItem(Item $cartItem): void
```
Comment (rogyar) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/UpdateCartItems.php:128]: Please note that all the validations and error handling are moved here https://github.com/magento/graphql-ce/pull/626/files#diff-23947b0596c6c269244b8667cb99db1aR46

**Comment by rogyar** [comment]
Api-functional tests for this feature are done. I will fix other api-functional tests soon.

**Comment by naydav** [comment]
Merged last mainline

**Comment by TomashKhamlai** [comment]
@rogyar could you please share your knowledge? What does it mean `@group recent`?

**Comment by TomashKhamlai** [comment]
Tried to update Virtual Product with multiple customizable option and something always goes wrong for me.  
The response is:
```graphql
{
  "errors": [
    {
      "message": "Could not update cart item: The product's required option(s) weren't entered. Make sure the options are entered and try again.",
      "category": "graphql-input",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "updateCartItems"
      ]
    }
  ],
  "data": {
    "updateCartItems": null
  }
}
```
Fixture `Magento/GraphQl/Catalog/_files/set_custom_options_simple_product.php` covers `area` and `field` customizable options.

It looks like multiple and checkbox are not covered.

**Comment by TomashKhamlai** [comment]
I found that after every update request that succeeded the `cart_item_id` is incremented. Please check that it is expected result of successful operation.

**Comment by rogyar** [comment]
Hi @TomashKhamlai. As for incremented `cart_item_id` - it's expected behavior as I described in the description. The same happens when you edit an item from the storefront. If an item's options were changed, the quote item is being recreated with a new set of options and a new `cart_item_id`. 
As for the multiselect options, I will check it out, thank you.

**Comment by rogyar** [comment]
Hi @TomashKhamlai. I've just added support for multi-select custom options as well

**Review by naydav — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @naydav, thank you for the review.
[ENGCOM-5349](https://jira.corp.magento.com/browse/ENGCOM-5349) has been created to process this Pull Request
:eight_spoked_asterisk: @naydav, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Pull request merged**